operator/v1/types_console.go

```diff
@@ -143,8 +143,94 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeType is the value of the CustomLogo Theme Type field that determines the theme mode in the console UI.
+type ThemeType string
+
+// ThemeType values
+const (
+	// ThemeTypeDark represents the dark theme for the console UI.
+	ThemeTypeDark ThemeType = "Dark"
+
+	// ThemeTypeLight represents the light theme for the console UI.
+	ThemeTypeLight ThemeType = "Light"
+
+	// ThemeTypeDefault represents the default theme for all unspecified themes.
+	ThemeTypeDefault ThemeType = "Default"
+)
+
+// LogoType is the value of the CustomLogo Type field that determines if the logo is for the masthead or the favicon in the console UI.
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// Theme defines a theme type for the console UI.
+type Theme struct {
+	// type is used to specify what theme(s) a logo will apply to in the console UI.
+	// type is a required field that allows values of Dark, Light, and Default.
+	// When set to Dark, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Dark theme.
+	// When set to Light, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Light theme.
+	// When set to Default, the logo file referenced in the 'file' field will be used for any themes types that do not have an explicit configuration.
+	// +kubebuilder:validation:Enum="Dark";"Light";"Default"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// file is used by the console to locate the specified file containing a custom logo.
+	// file is a required field that references a ConfigMap name and key that contains the custom logo file in the openshift-config namespace.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file'
+	// The ConfigMap key must include the file extension for the dark, light, or default theme so that the console serves the file with the correct MIME type, which can be '.svg', '.png', or whatever the browser supports.
+	// Recommended Masthead and About Modal logo specifications:
+	// Image size must be less than 1 MiB due to constraints on the ConfigMap size.
+	// Dimensions: Max height of 60px and max width of 200px.
+	// SVG format preferred.
+	// Recommended Favicon specifications:
+	// Dimensions: Max height of 16px and max width of 16px
+	// PNG format preferred.
+	// +required
+	File configv1.ConfigMapFileReference `json:"file"`
```

Comment:
Create your own reference type, don't use someone elses

If they add a new field, you start supporting it whether you meant to or not